src/CosmosDB/CosmosDB/CosmosDBAccount/NewAzCosmosDBAccount.cs

```diff
@@ -0,0 +1,193 @@
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Management.Automation;
+using Microsoft.Azure.Commands.CosmosDB.Helpers;
+using Microsoft.Azure.Commands.CosmosDB.Models;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Commands.ResourceManager.Common.Properties;
+using Microsoft.Azure.Management.CosmosDB;
+using Microsoft.Azure.Management.CosmosDB.Models;
+
+namespace Microsoft.Azure.Commands.CosmosDB
+{
+    [Cmdlet(VerbsCommon.New, ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "CosmosDBAccount", DefaultParameterSetName = NameParameterSet, SupportsShouldProcess = true), OutputType(typeof(PSDatabaseAccount))]
+    public class NewAzCosmosDBAccount : AzureCosmosDBCmdletBase
+    {
+        [Parameter(Mandatory = true, ParameterSetName = NameParameterSet, HelpMessage = Constants.ResourceGroupNameHelpMessage)]
+        [ResourceGroupCompleter]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(Mandatory = true, ParameterSetName = NameParameterSet, HelpMessage = Constants.AccountNameHelpMessage)]
+        public string Name { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = Constants.DefaultConsistencyLevelHelpMessage)]
+        [PSArgumentCompleter("BoundedStaleness", "ConsistentPrefix", "Eventual", "Session", "Strong")]
+        public string DefaultConsistencyLevel { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = Constants.EnableAutomaticFailoverHelpMessage)]
+        public SwitchParameter EnableAutomaticFailover { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = Constants.EnableMultipleWriteLocationsHelpMessage)]
+        public SwitchParameter EnableMultipleWriteLocations { get; set; }
+
+        [Parameter(Mandatory = false, ParameterSetName = NameParameterSet, HelpMessage = Constants.EnableVirtualNetworkHelpMessage)]
+        public SwitchParameter EnableVirtualNetwork { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = Constants.IpRangeFilterHelpMessage)]
+        public string[] IpRangeFilter { get; set; }
+
+        [Parameter(Mandatory = true, HelpMessage = Constants.LocationHelpMessage)]
+        public string[] Location { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = Constants.MaxStalenessIntervalInSecondsHelpMessage)]
+        public int? MaxStalenessIntervalInSeconds { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = Constants.MaxStalenessPrefixHelpMessage)]
+        public int? MaxStalenessPrefix { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = Constants.TagHelpMessage)]
+        public Hashtable Tag { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = Constants.VirtualNetworkRuleHelpMessage)]
+        public string[] VirtualNetworkRule { get; set; }
```

Comment:
In the CLI we accept a string array because of limitations for processing input. Could you have object inputs in the command module? Or do we want to keep it basic? For example, a VirtualNetworkRule object contains both the "id" and a Boolean "ignoreMissingVnetServiceEndpoint". Could we allow users to set both?

This is similar for the "Locations" input. In the CLI we allow setting "locationName", "failoverPriority", and "isZoneRedundant". (Also should these array parameter names be plural?)

Reply:
Thanks for pointing this out sam. I have added two new PSObjects: PSLocation and PSVirtualNetworkRule to handle this. User can enter either string or as an object now. 

The array names arent plural to stay in alignment with Powershell naming conventions. Anything which can have a single value has to have a singular name in Powershell.